test/CommandLineUtils.Tests/ValueParserProviderTests.cs

```diff
@@ -130,6 +144,76 @@ public void ParsesLongNullable(string arg, long? result)
             Assert.Equal(result, parsed.Int64Opt);
         }
 
+        [Theory]
+        [InlineData("0.0", 0.0f)]
+        [InlineData("123456789.987654321", 123456789.987654321f)]
+        [InlineData("-123.456", -123.456f)]
+        [InlineData("-1E10", -1E10f)]
+        [InlineData("-Infinity", float.NegativeInfinity)]
+        [InlineData("Infinity", float.PositiveInfinity)]
+        [InlineData("NaN", float.NaN)]
+        public void ParsesFloat(string arg, float result)
+        {
+            // can't work out how to set a string representation of infinity, in an attribute, for current culture
```

Comment:
You probably would need to use `MemberData` instead, and return this from a property

Reply:
Ah right. I've not used xunit before. I'll fix it up

Reply:
Here's a good example:
```c#
public class CalculatorTests  
{
    [Theory]
    [MemberData(nameof(GetData), parameters: 3)]
    public void CanAddTheoryMemberDataMethod(int value1, int value2, int expected)
    {
        var calculator = new Calculator();

        var result = calculator.Add(value1, value2);

        Assert.Equal(expected, result);
    }

    public static IEnumerable<object[]> GetData(int numTests)
    {
        var allData = new List<object[]>
        {
            new object[] { 1, 2, 3 },
            new object[] { -4, -6, -10 },
            new object[] { -2, 2, 0 },
            new object[] { int.MinValue, -1, int.MaxValue },
        };

        return allData.Take(numTests);
    }
}
```
https://andrewlock.net/creating-parameterised-tests-in-xunit-with-inlinedata-classdata-and-memberdata/